# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/brush_mix_blend.glsl
# Commit: ee0de0eac18d
# Full Hash: ee0de0eac18d1ba825467f6629ded5e613c56627
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 1 - Remove PrevPassAlpha and PrevPassColor r=nical
#   
#   Remove usage of the implicit prev pass alpha and color texture
#   samplers from batching / renderer / shader code. They are replaced
#   by explicit references to the texture ID for the source task.
# ==============================================================================

diff -r ecf2ab52eb03 -r ee0de0eac18d gfx/wr/webrender/res/brush_mix_blend.glsl
--- a/gfx/wr/webrender/res/brush_mix_blend.glsl	Thu Dec 17 22:10:02 2020 +0000
+++ b/gfx/wr/webrender/res/brush_mix_blend.glsl	Thu Dec 17 22:26:08 2020 +0000
@@ -32,7 +32,8 @@
 ) {
     //Note: this is unsafe for `vi.world_pos.w <= 0.0`
     vec2 device_pos = vi.world_pos.xy * pic_task.device_pixel_scale / max(0.0, vi.world_pos.w);
-    vec2 texture_size = vec2(textureSize(sPrevPassColor, 0));
+    vec2 backdrop_texture_size = vec2(textureSize(sColor0, 0));
+    vec2 src_texture_size = vec2(textureSize(sColor1, 0));
     v_op = prim_user_data.x;
 
     PictureTask src_task = fetch_picture_task(prim_user_data.z);
@@ -40,7 +41,7 @@
     vec2 src_uv = src_device_pos +
                   src_task.common_data.task_rect.p0 -
                   src_task.content_origin;
-    v_src_uv.xy = src_uv / texture_size;
+    v_src_uv.xy = src_uv / src_texture_size;
     v_src_uv.z = src_task.common_data.texture_layer_index;
 
     RenderTaskCommonData backdrop_task = fetch_render_task_common_data(prim_user_data.y);
@@ -48,7 +49,7 @@
     vec2 backdrop_uv = device_pos +
                        backdrop_task.task_rect.p0 -
                        src_task.content_origin * src_to_backdrop_scale;
-    v_backdrop_uv.xy = backdrop_uv / texture_size;
+    v_backdrop_uv.xy = backdrop_uv / backdrop_texture_size;
     v_backdrop_uv.z = backdrop_task.texture_layer_index;
 }
 #endif
@@ -212,8 +213,8 @@
 const int MixBlendMode_Luminosity  = 15;
 
 Fragment brush_fs() {
-    vec4 Cb = textureLod(sPrevPassColor, vec3(v_backdrop_uv.xy, v_backdrop_uv.z), 0.0);
-    vec4 Cs = textureLod(sPrevPassColor, vec3(v_src_uv.xy, v_src_uv.z), 0.0);
+    vec4 Cb = textureLod(sColor0, vec3(v_backdrop_uv.xy, v_backdrop_uv.z), 0.0);
+    vec4 Cs = textureLod(sColor1, vec3(v_src_uv.xy, v_src_uv.z), 0.0);
 
     // The mix-blend-mode functions assume no premultiplied alpha
     if (Cb.a != 0.0) {